### DISCUSSION TITLE
Creating viewset actions on the fly

### DISCUSSION THREAD

**Discussion by iamMHZ** [encode/django-rest-framework#9008] [Ideas & Suggestions] [answered]
Most of the time a view set action contains lots of boilerplate code. Please take a look at the example below.

```
@action(detail=True, methods=['patch'], url_path='change-password', url_name='change_password')
    def change_password(self, request, **kwargs):
        instance = self.get_object()
        serializer = UserChangePasswordSerializer(instance=instance)
        serializer.is_valid(raise_exception=True)
        serializer.save()
        return Response()
```

It would be really nice if we can have this option to define a viewset actions like this:
```
change_password = Action(serializer =UserChangePasswordSerializer,detail=True, methods=['patch'], url_path='change-password', url_name='change_password') 
```
It removes lots of boilerplate code and passes the actual duty to the seializer.

**Comment by Ehsan200** [comment, marked as answer]
Hello there!
DRF provides some general actions itself to prevent you from doing certain things in your APIView. By using the action decorator, you can add a new custom action to your viewset, which requires specifying its complete functionality. However, in this case, you are only performing a partial_update on your user, which can be accomplished without adding a new action.
